data_validation/gcs_helper.py

```diff
@@ -0,0 +1,93 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import os
+from typing import List
+
+from google.cloud import storage
+from data_validation import client_info
+
+
+def _is_gcs_path(file_path: str) -> bool:
+    return True if file_path.startswith("gs://") else False
+
+
+def _get_validation_path(name: str) -> str:
+    """Returns the full path to a validation."""
+    if _is_gcs_path(name):
+        return name
+    return os.path.join("./", name)
+
+
+def _get_gcs_bucket(gcs_file_path: str) -> storage.Bucket:
+    """Returns storage.Bucket given GCS file path with prefix."""
+    bucket_name = gcs_file_path[5:].split("/")[0]
+    info = client_info.get_http_client_info()
+    storage_client = storage.Client(client_info=info)
+    try:
+        return storage_client.bucket(bucket_name)
+    except ValueError as e:
+        raise ValueError("GCS Path Failure {} -> {}".format(gcs_file_path, e))
+
+
+def _get_gcs_file_path(gcs_file_path: str) -> str:
+    """
+    Returns relative object file path i.e. `path/to/file.yaml`
+    given full GCS file path with prefix.
+    """
+    return str.join("", gcs_file_path[5:].split("/", 1)[1:])
```

Comment:
nit: I find joining to the string itself rather than via `str` neater, for example:
```
"".join(gcs_file_path[5:].split("/", 1)[1:])
```

But this is just preference so feel free to ignore.

Reply:
I like this a lot better!